nfs/src/rpc_task.cpp

```diff
@@ -978,22 +1006,23 @@ static void sync_membuf(const struct bytes_chunk& bc,
      * Note that we allocate the rpc_task for flush before the lock as it may
      * block.
      */
-    if (!mb->is_dirty() || mb->is_flushing()) {
+    if (!mb->is_dirty() || (mb->is_flushing() && rpc_api->pvt == nullptr)) {
+        free_rpc_task();
         return;
     }
 
-    flush_task = client->get_rpc_task_helper()->alloc_rpc_task(FUSE_FLUSH);
-    flush_task->init_flush(nullptr /* fuse_req */, ino);
-    flush_task->set_op_type(FUSE_FLUSH);
-
```

Comment:
it doesn't look clean to force every caller to create a task and then call sync_membuf() on that.
If it's only for handling the possibility of ctx being set in the case of jukebox, we can have ctx as an extra argument to sync_membuf()

Reply:
No Tomar, I need to do it for JukeBox, nfs_client create a task, then for that task can call sync_membuf. same thing we do for read also.

Reply:
I know we do it for read, but for read the func is read_from_server() which kind of looks ok to be operating on a task, as a task would tell it what to read (and the bc argument tells it where to read).
In here the function is sync_membuf() so feels appropriate that it takes the membuf and syncs it (allocating anything needed to achieve that).
Let me see.